variantlib/commands/make_variant.py

```diff
@@ -179,28 +185,27 @@ def make_variant(args: list[str]) -> None:
         if not (metadata_f := distinfo_dir / "METADATA").exists():
             raise FileNotFoundError(metadata_f)
 
-        with metadata_f.open(mode="r+") as file:
-            # Read all lines
-            lines = file.readlines()
+        with metadata_f.open(mode="r+b") as file:
+            # Parse the metadata
+            metadata_parser = email.parser.BytesParser()
+            metadata = metadata_parser.parse(file)
 
-            # Remove trailing empty lines
-            while lines and lines[-1].strip() == "":
-                lines.pop()
+            # Replace the variant headers
+            del metadata["Variant"]
```

Comment:
Question:
If we `make_variant` an already-variant wheel.

Do you expect to "append new properties" or "replace them" with new properties ?

Reply:
The existing code replaced them, so I followed suit. I can add an option to append instead, but then we'd have to read properties from input, update their values (if we specify the same feature as found in input) and reconstruct `VariantDescription`. Then we also open the can of worms called "should we have an option to delete some property?"